packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -360,45 +359,46 @@ export const thorchainApi: SwapperApi = {
       const thorTxHash = txHash.replace(/^0x/, '')
 
       // not using monadic axios, this is intentional for simplicity in this non-monadic context
-      const { data } = await axios.get<ThornodeStatusResponse>(
-        `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/status/${thorTxHash}`,
-      )
-
-      if ('error' in data) {
+      const [{ data: txData }, { data: txStatusData }] = await Promise.all([
+        axios.get<ThornodeTxResponse>(
+          `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/${thorTxHash}`,
+        ),
+        axios.get<ThornodeStatusResponse>(
+          `${config.REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/tx/status/${thorTxHash}`,
+        ),
+      ])
+
+      // We care about txStatusData errors because it drives all of the status logic.
+      if ('error' in txStatusData) {
         return {
           buyTxHash: undefined,
           status: TxStatus.Unknown,
           message: undefined,
         }
       }
 
-      const latestOutTx = data.out_txs?.[data.out_txs.length - 1]
-      const hasOutboundTx = latestOutTx?.chain !== 'THOR'
+      // We use planned_out_txs to determine the number of out txs because we don't want to derive
+      // swap completion based on the length of out_txs which is populated as the trade executed
+      const numOutTxs = txStatusData.planned_out_txs?.length ?? 0
+      const lastOutTx = txStatusData.out_txs?.[numOutTxs - 1]
 
-      const buyTxHash = parseThorBuyTxHash(txHash, latestOutTx)
+      const buyTxHash = parseThorBuyTxHash(txHash, lastOutTx)
 
-      // if we have an outbound transaction (non rune) and associated buyTxHash, check if it's been confirmed on-chain
-      if (hasOutboundTx && buyTxHash) {
-        const outboundTxConfirmations = await checkOutboundTxConfirmations(
+      // If thornode says the tx is done, its done ;)
+      // We discard txData errors because we can derive the status before it returns a success response.
+      if (!('error' in txData) && txData.observed_tx?.status === 'done') {
```

Comment:
q: Do we want to early return on `error` in `txData` the same way we do for `txStatusData`? Assuming this would mostly be a paranoia check because if there is an error in the Tx, it would most likely be caught in the status error checks above, but would make things some cleaner

Reply:
The concern I have is that the txData fetch can fail if it has not been seen by that endpoint for any reason. Ideally we're still able to determine the tx status in the absence of txData

Reply:
Agree it would make things cleaner for sure

Reply:
Sounds sane to me! Think we can simply update the comment to disambiguate and say precisely that i.e we want to *ensure* there is no error in txData vs. *discarding* it, but continue and assume pending if `{error: string}` is returned